post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java

```diff
@@ -131,25 +132,42 @@ protected void processCachedIds() {
                         processTopic(keyTopic, entity, ids, patientRepository::executeStoredProcForPatientIds);
                         break;
                     case INVESTIGATION:
-                        List<InvestigationResult> invData = processTopic(keyTopic, entity, ids, investigationRepository::executeStoredProcForPublicHealthCaseIds);
+                        List<InvestigationResult> invData = processTopic(keyTopic, entity, ids,
+                                investigationRepository::executeStoredProcForPublicHealthCaseIds);
                         ids.forEach(id -> {
                             if (idVals.containsKey(id)) {
-                                processId(id, idVals.get(id), investigationRepository::executeStoredProcForPageBuilder, Entity.CASE_ANSWERS);
+                                processId(id, idVals.get(id),
+                                        investigationRepository::executeStoredProcForPageBuilder, Entity.CASE_ANSWERS);
                                 idVals.remove(id);
                             }
                         });
-                        processTopic(keyTopic, Entity.F_PAGE_CASE, ids, investigationRepository::executeStoredProcForFPageCase);
+                        processTopic(keyTopic, Entity.F_PAGE_CASE, ids,
+                                investigationRepository::executeStoredProcForFPageCase);
                         datamartProcessor.process(invData);
```

Comment:
Now that I am thinking about it, in the future if we need to validate that there is no lag between the nrt_investigation topic and nbs_datamart datamart topic.

Reply:
What do you think we need to do for this. Do we need to do a coupling between investigation and datamart topics.